src/ucs/sys/compiler_def.h

```diff
@@ -78,7 +78,7 @@
 #define UCS_BIT(i)               (1ul << (i))
 
 /* Mask of bits 0..i-1 */
-#define UCS_MASK(_i)             (((_i) >= 64) ? ~0 : (UCS_BIT(_i) - 1))
+#define UCS_MASK(_i)             (((_i) >= 64) ? ~0ul : (UCS_BIT(_i) - 1))
```

Comment:
I think this one is intentional, @ivankochin/@yosefe?

Reply:
yes, to avoid compiler warnings

Reply:
Not on a tertiary, the compiler is promoting everything to `ul` anyway (even when it is not complaining about it).

Reply:
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=81413&view=logs&j=b225495a-a138-58be-0e0c-72778e1ff6b6&t=c6c5013e-bc26-58e0-88ac-2d26030ca3ad
```
/__w/1/s/contrib/../src/uct/ib/mlx5/ib_mlx5.c:218:26: error: implicit conversion from 'unsigned long' to 'unsigned int' changes value from 18446744073709551615 to 4294967295 [-Werror,-Wconstant-conversion]
    cq->cq_length_mask = UCS_MASK(cq->cq_length_log);
                       ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/1/s/contrib/../src/ucs/sys/compiler_def.h:81:50: note: expanded from macro 'UCS_MASK'
#define UCS_MASK(_i)             (((_i) >= 64) ? ~0ul : (UCS_BIT(_i) - 1))
                                                 ^~~~
1 error generated.
```

Reply:
Interesting, the DPC++ compiler choke on this. Without the `ul` I get this from gcc 
```
.../include/ucs/sys/compiler_def.h:81:50: warning: operand of '?:' changes signedness from 'int' to 'long unsigned int' due to unsignedness of other operand [-Wsign-compare]
   81 | #define UCS_MASK(_i)             (((_i) >= 64) ? ~0 : (UCS_BIT(_i) - 1))
      |                                                  ^~
.../include/ucp/api/ucp.h:511:31: note: in expansion of macro 'UCS_MASK'
  511 |     UCP_DATATYPE_CLASS_MASK = UCS_MASK(UCP_DATATYPE_SHIFT) /**< Data-type class
```

Reply:
We can keep both compilers happy if we use `~0u`. Ugly, but it works, there is no demotion of int and no signedness mismatch.

Reply:
```
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/mem_buffer.cc:321: Failure
Pattern check failed at 0x289dd10 offset 0: Expected: 0x11111111 Actual: 0x1111111111111111
```

It's related to UCS_MASK change I guess. We invert unsigned 32bits, **then** convert to 64bits.

Reply:
This is puzzling. The current version of the macro does exactly as before, except in unsigned instead of not specified. I'm confused about how did it work before. Was the compiler propagating the type of the lvalue to the 0 in the macro ?

I guess we'll live with the warnings.